yarn-project/prover-client/src/orchestrator/block-building-helpers.ts

```diff
@@ -515,6 +517,15 @@ export async function executeBaseRollupCircuit(
 ): Promise<[BaseOrMergeRollupPublicInputs, Proof]> {
   logger?.(`Running base rollup for ${tx.hash}`);
   const rollupOutput = await simulator.baseRollupCircuit(inputs);
+  const txEffect = toTxEffect(tx);
+  const txEffectHash = txEffect.hash();
+  if (!rollupOutput.txsEffectsHash.toBuffer().equals(txEffectHash)) {
```

Comment:
Just flagging: I believe at present this would kill the whole proving ticket, which is probably not the behavior we want when one TX is invalid.